### ISSUE TITLE
Rollback the stack if UserData failed

### ISSUE THREAD

**Issue by ChristopheBougere** [PokaInc/cloudformation-ami#2]
Hi,
I think it could be interesting to add different values in the `UserDataFinished` tag to know if the UserData succeeded or failed.
That way, we could rollback the stack if the user data failed, instead of updating it with a corrupted AMI.
A way to achieve it in the UserData script is to assign a variable to 0 at the beginning of the script (`scriptFailed=0`), and adding ` || scriptFailed=1` at the end of every line.
What do you think?

**Comment by spg** [comment]
Yes that seems like a good idea to me. At the moment, if the User Data script fails, the `UserDataFinished` tag will never be set to `true`, causing the State Machine to loop forever.
Detecting failures would indeed help getting out of that infinite loop.
